core/src/test/java/com/linecorp/armeria/server/ServerTest.java

```diff
@@ -308,6 +310,17 @@ public void testUnsupportedMethod() throws Exception {
         testSimple("WHOA / HTTP/1.1", "HTTP/1.1 405 Method Not Allowed");
     }
 
+    @Test
+    public void duplicatedPort() {
+        Server duplicatedPortServer = new ServerBuilder()
+                .http(server.httpPort())
+                .service("/", (ctx, res) -> HttpResponse.of(""))
+                .build();
+        assertThatThrownBy(() -> duplicatedPortServer.start().join())
+                .hasCauseInstanceOf(BindException.class)
+                .hasMessageContaining("Address already in use");
```

Comment:
I guess this may fail because I/O exception messages are often platform dependent.